pom.xml

```diff
@@ -933,6 +957,34 @@
                 <artifactId>jucx</artifactId>
                 <version>${ucx.version}</version>
             </dependency>
+            <!-- -->
+            <dependency>
+                <groupId>org.slf4j</groupId>
+                <artifactId>slf4j-api</artifactId>
+                <version>${slf4j.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-slf4j2-impl</artifactId>
+                <version>${log4j.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-api</artifactId>
+                <version>${log4j.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-core</artifactId>
+                <version>${log4j.version}</version>
+            </dependency>
+            <dependency>
+                <!-- API bridge between log4j 1 and 2 -->
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-1.2-api</artifactId>
+                <version>${log4j.version}</version>
+            </dependency>
+            <!-- -->
```

Comment:
Curious why all of these were added?  Especially wondering about the log4j dependencies since we normally want to use slf4j APIs when logging instead of hitting log4j.

Normally we leave these sort of dependencies out because we explicitly want to pull in whatever version Spark is using and compile against that.  The risk of explicitly pulling this in is that it might conflict with the version used by the particular Spark version we're compiling against and instead of a compile time problem we get a runtime problem.

Reply:
I added this because I encountered errors related to log4j & slf4j such as `NoSuchMethod` when running unit tests.  After adding them, they were gone. I agree that it might introduce risks for this approach. Any better approaches we can try?

Reply:
That seems to imply there's an issue with the test classpath where the APIs that were available at compile-time aren't present at runtime.  I'd have Maven dump the compile and test classpaths and see if you can find jars that are missing in test vs. compile that would explain it.  We should be picking these up as transitive dependencies from the Spark jars.  Maybe there's an issue with the EMR Spark dependencies where that's somehow not the case (but only at test runtime?!).